Building lldb-server for Chrome OS requires several pieces of clang &
llvm to be cross-compiled.  This is expensive and time-consuming, and
does not need to be done repeatedly.  This patch updates the
lldb-server cmakefiles to check to see if the required pieces have
already been built and to not create a native build when
LLDB_TABLEGEN_EXE is already provided.

Author: manojgupta@google.com
diff --git a/lldb/CMakeLists.txt b/lldb/CMakeLists.txt
index 11329052d0b8..3457b7680d19 100644
--- a/lldb/CMakeLists.txt
+++ b/lldb/CMakeLists.txt
@@ -63,7 +63,7 @@ if(LLVM_ENABLE_MODULES)
   list(APPEND LLVM_COMMON_DEPENDS intrinsics_gen)
 endif()
 
-if(CMAKE_CROSSCOMPILING AND LLDB_BUILT_STANDALONE)
+if(CMAKE_CROSSCOMPILING AND LLDB_BUILT_STANDALONE AND NOT LLDB_TABLEGEN_EXE)
   set(LLVM_USE_HOST_TOOLS ON)
   include(CrossCompile)
   if (NOT NATIVE_LLVM_DIR OR NOT NATIVE_Clang_DIR)
